flow/activities/flowable.go

```diff
@@ -45,9 +45,10 @@ type SlotSnapshotSignal struct {
 type FlowableActivity struct {
 	CatalogPool *pgxpool.Pool
 	Alerter     *alerting.Alerter
+	CdcCacheRw  sync.RWMutex
```

Comment:
Maybe have a replication connection manager struct that takes care of:
1. This map and locking.
2. Keeping track of the connection health and lifecycle.
3. Any other additional metadata pertaining to the replication connection.

Reply:
Tracking connection health & lifecycle are part of `MaintainPull` which needs to exist either way to keep heartbeating session